src/software/ai/hl/stp/play/play.cpp

```diff
@@ -137,3 +310,56 @@ void Play::updateTactics(const PlayUpdate &play_update)
 {
     play_update.set_tactics(getTactics(play_update.world));
 }
+
+std::unique_ptr<TbotsProto::PrimitiveSet> Play::getPrimitivesFromTactic(
+    const GlobalPathPlannerFactory &path_planner_factory, const World &world,
+    std::shared_ptr<Tactic> tactic,
+    std::set<TbotsProto::MotionConstraint> motion_constraints) const
+{
+    auto path_planner = path_planner_factory.getPathPlanner(motion_constraints);
+    CreateMotionControl create_motion_control =
+        [path_planner, motion_constraints](const Robot &robot, const Point &destination) {
+            Point robot_position = robot.position();
+            TbotsProto::MotionControl motion_control;
+            TbotsProto::Path path_proto;
+
+            // first point is always the robot_position
+            std::vector<Point> path_points = {robot_position, robot_position};
+            auto path = path_planner->findPath(robot_position, destination);
+            *(motion_control.mutable_requested_destination()) =
+                *createPointProto(destination);
+
+            if (path.has_value() && path.value().getKnots().size() > 1)
+            {
+                path_points = path.value().getKnots();
+                motion_control.set_normalized_path_length(
+                    EnlsvgPathPlanner::pathLength(path_points, robot_position) /
+                    EnlsvgPathPlanner::MAX_PATH_LENGTH);
+            }
+            else
+            {
+                LOG(WARNING) << "No path found for robot " << robot.id() << " to "
+                             << destination << std::endl;
+                motion_control.set_normalized_path_length(1.0);
+            }
+
+            CHECK(path_points.size() >= 2)
+                << "Empty path: " << path_points.size() << std::endl;
```

Comment:
this is not true, there should never be empty paths since the path defaults to the current robot position, hence your complaint about the log spam, but no complaint about CHECKs trigger :smile: 
that said, the check isn't really necessary